cpp/src/arrow/type_traits.h

```diff
@@ -942,6 +991,42 @@ static inline bool is_primitive(Type::type type_id) {
   return false;
 }
 
+/// \brief Check for a primitive-like type
+///
+/// \param[in] type_id the type-id to check
+/// \return whether type-id is a primitive-like type one
+static inline bool is_primitive_like(Type::type type_id) {
```

Comment:
Would `is_not_nested` or something be clearer here? "primitive-like" is a little confusing (though, I suppose that then raises the question of why fixed-width binary and decimal are also missing here)

Reply:
I forgot to remove this predicate and I'll do so soon. It was only added to correspond to some `*Types()` function, all of which are going to be removed.

Reply:
Side note: the existing (pre-PR) code has an `is_nested` predicate, though it is not the complement of the function discussed here, e.g., due to timestamp and interval types.